onnxruntime/core/providers/rocm/math/softmax.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "softmax.h"
+#include "core/providers/common.h"
+#include "core/providers/rocm/miopen_common.h"
+
+namespace onnxruntime {
+namespace rocm {
+
+template <typename T, bool is_log_softmax>
+Status SoftMaxComputeHelper(
+    const T* X,
+    const TensorShape& input_shape,
+    T* Y,
+    miopenHandle_t handle,
+    int64_t axis) {
+  typedef typename ToHipType<T>::MappedType HipT;
+
+  const int64_t normalized_axis = HandleNegativeAxis(axis, input_shape.NumDimensions());
+
+  int64_t N = input_shape.SizeToDimension(normalized_axis);
+  int64_t D = input_shape.SizeFromDimension(normalized_axis);
+  auto Y_data = reinterpret_cast<HipT*>(Y);
+  auto X_data = reinterpret_cast<const HipT*>(X);
+
+  // miopenSoftmaxForward/Backward is not optimal implementation.
```

Comment:
Is this true for miopen case as well? I don't understand why these libraries can't special case for small sizes.

Reply:
Didn't verify it. Just created an task to track it.